samcli/commands/_utils/options.py

```diff
@@ -53,6 +53,13 @@ def get_or_default_template_file_name(ctx, param, provided_value, include_build)
         # sam configuration file should always be relative to the supplied original template and should not to be set
         # to be .aws-sam/build/
         setattr(ctx, "samconfig_dir", os.path.dirname(original_template_path))
+        try:
+            template_data = get_template_data(result)
+            setattr(ctx, "template_dict", template_data)
```

Comment:
We should not be doing setattr directly. Can we not follow the same pattern as profile or region which have explicit setter methods?

Reply:
https://github.com/aws/aws-sam-cli/blob/develop/samcli/cli/options.py#L33, something like that makes sense.